app/src/main/java/com/globant/harrypotterapp/fragment/HouseDetailFragment.kt

```diff
@@ -9,49 +9,59 @@ import android.widget.Toast
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = FragmentHouseDetailBinding.inflate(layoutInflater)
         houseDetailViewModel.getHouseDetailLiveData().observe(::getLifecycle, ::updateUI)
-        houseDetailViewModel.fetchHouseDetail()
     }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
         binding.root
 
-    private fun updateUI(houseDetail: HouseDetail) {
-        houseDetail.apply {
-            binding.houseDetailFragmentName.text = getString(R.string.house_detail_fragment_name_text, this.name)
-            binding.houseDetailFragmentMascot.text = getString(R.string.house_detail_fragment_mascot_text, this.mascot)
-            binding.houseDetailFragmentHeadOfHouse.text = getString(R.string.house_detail_fragment_head_of_house_text, this.headOfHouse)
-            binding.houseDetailFragmentHouseGhost.text = getString(R.string.house_detail_fragment_ghost_text, this.houseGhost)
-            binding.houseDetailFragmentFounder.text = getString(R.string.house_detail_fragment_founder_text, this.founder)
-            when (this.name) {
+    private fun updateUI(data: Event<Data<List<HouseDetail>>>) {
+        val houseDetailData = data.getContentIfNotHandled()
+        when (houseDetailData?.status) {
+            Status.LOADING -> binding.houseDetailFragmentLoader.visibility = View.VISIBLE
+            Status.SUCCESS -> showHouseDetailData(houseDetailData.data?.get(ZERO))
+            Status.ERROR -> showErrorMessage()
+        }
+    }
+
+    private fun showHouseDetailData(houseDetail: HouseDetail?) {
+        binding.houseDetailFragmentLoader.visibility = View.GONE
+        houseDetail?.apply {
+            binding.houseDetailFragmentName.text = getString(R.string.house_detail_fragment_name_text, name)
+            binding.houseDetailFragmentMascot.text = getString(R.string.house_detail_fragment_mascot_text, mascot)
+            binding.houseDetailFragmentHeadOfHouse.text = getString(R.string.house_detail_fragment_head_of_house_text, headOfHouse)
+            binding.houseDetailFragmentHouseGhost.text = getString(R.string.house_detail_fragment_ghost_text, houseGhost)
+            binding.houseDetailFragmentFounder.text = getString(R.string.house_detail_fragment_founder_text, founder)
+            when (name) {
                 GRYFFINDOR -> binding.houseDetailFragmentImage.setImageResource(R.drawable.gryffindor)
                 RAVENCLAW -> binding.houseDetailFragmentImage.setImageResource(R.drawable.ravenclaw)
                 SLYTHERIN -> binding.houseDetailFragmentImage.setImageResource(R.drawable.slytherin)
                 HUFFLEPUFF -> binding.houseDetailFragmentImage.setImageResource(R.drawable.hufflepuff)
-                else -> {
-                    binding.houseDetailFragmentImage.setImageResource(R.drawable.hogwarts)
-                    showErrorMessage()
```

Comment:
shouldnt you call `showErrorMessage` on else branch

Reply:
Yes, done!